pkgs/applications/misc/gcstar/default.nix

```diff
@@ -0,0 +1,76 @@
+{ stdenv
+, fetchFromGitLab
+, bash
+, perl
+, gtk2
+, perlPackages
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gcstar";
+  version = "1.7.2";
+
+  src = fetchFromGitLab {
+      owner = "Kerenoc";
+      repo = "GCstar";
+      rev = "v${version}";
+      sha256 = "1vqfff33sssvlvsva1dflggmwl00j5p64sn1669f9wrbvjkxgpv4";
+    };
+
+  perlDeps = [
+    perl
```

Comment:
I would worry about coherence and either use `perl` + `perl.pkgs.*`, or `perlPackages.perl` + `perlPackages.*`.

Reply:
I am afraid I don't get this one... I am not that strong at syntax... do you mean something like

`perlDeps = perl + perlPackages [ package1 package2 .... ]`

(I know that it is syntactically wrong but don't know how to fix it off the top of my head)

Reply:
I wrote it this way:
```

perlDeps = [ perl ] ++ (with perlPackages; [
    ArchiveZip
    DateCalc
    DateTimeFormatStrptime
    Glib
    Gtk2
    GD
    GDGraph
    GDText
    HTMLParser
    JSON
    ImageExifTool
    librelative
    LWPUserAgent
    LWPProtocolHttps
    MP3Info
    MP3Tag
    NetFreeDB
    OggVorbisHeaderPurePerl
    Pango
    XMLSimple
    XMLParser
  ]);
```

Reply:
Sorry, I just meant using `perl` from `perlPackages` or using packages from `perl.pkgs` to avoid mismatches when someone overrides just one of those.

Reply:
Now it makes use of `perl` from `perlPackages`